daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -369,10 +367,10 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
           )
 
         case SResultNeedTime(callback) =>
-          onLedger.dependsOnTime = true
           callback(time)
 
         case SResultNeedKey(gk, _, cb) =>
+          onLedger.dependsOnTime = true
```

Comment:
this is wrong, I liked the change tbh but if you want to revert it that’s fine, just revert it properly :)

Reply:
OK the response to your question  
> Why don’t we modify [`dependsOnTime`] in speedy?
is that `SBGetTime` can be called in `offLedger` Mode (in `ScenarioRunner`).

Two solutions: 
- let the engine modifies the flag how it was before 14d7d80b53880f4f99dd0ffee98323c397eb2d6c
- modify conditionally the flag in SGetTime e6a1340465dcb2afa605c572aec6884f6b6468d4

Reply:
I’d go for conditionally modifying the flag in SGetTime but happy to leave the choice to you.

Reply:
I am not sure to understand if e6a1340465dcb2afa605c572aec6884f6b6468d4 (the top of my PR) is fine or not ?

Reply:
that looks good, I was still looking at the commit before that

Reply:
:bow: